infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -226,6 +231,8 @@
                         .setNameFormat(serverContext.getThreadPrefix() + "LocalMetricsPolling")
                         .build());
 
+        this.sequencerRecoveryFuture = CompletableFuture.completedFuture(true);
```

Comment:
you just care about this on init, can we remove these futures they are very confusing.